inc/html.class.php

```diff
@@ -6688,74 +6692,57 @@ public static function compileScss($args) {
          );
       }
 
-      if (!isset($args['file']) || $args['file'] == 'main_styles') {
-         $files[] = 'css/styles';
-         if (isset($_SESSION['glpihighcontrast_css'])
-            && $_SESSION['glpihighcontrast_css']) {
-            $ckey .= '_highcontrast';
-            $files[] = 'css/highcontrast';
-         }
+      $file = isset($args['file']) ? $args['file'] : 'css/styles';
 
-         // CSS theme
-         $theme = 'auror';
-         if (isset($_SESSION["glpipalette"])) {
-            $theme = $_SESSION['glpipalette'];
-         }
-         $ckey .= '_' . $theme;
-         $files[] = 'css/palettes/' . $theme;
-      } else {
-         $ckey .= '_' . md5($args['file']);
+      $ckey .= '_' . md5($file);
+      $ckey = 'css_' . md5($ckey);
 
-         $filename = realpath(GLPI_ROOT . '/' . $args['file']);
-         if (!Toolbox::startsWith($filename, realpath(GLPI_ROOT))) {
-            // Prevent import of a file from ouside GLPI dir
-            return '';
-         }
+      if (!Toolbox::endsWith($file, '.scss')) {
+         // Prevent include of file if ext is not .scss
+         $file .= '.scss';
+      }
 
-         if (!Toolbox::endsWith($args['file'], '.scss')) {
-            // Prevent include of file if ext is not .scss
-            $args['file'] .= '.scss';
-         }
+      // Requested file path
+      $path = GLPI_ROOT . '/' . $file;
+      $path = realpath($path);
```

Comment:
relatpath added ?

Reply:
I moved it [here](https://github.com/glpi-project/glpi/pull/5872/files#diff-298092ebe134c5b9ccbc7042328fb02cR6722). It is used to check that requested file is inside GLPI tree.